docs/design/disqualification.md

```diff
@@ -0,0 +1,77 @@
+
+'Failing to return data' is clarified to mean during an audit or a repair. Failure to return data to uplinks is specifically excluded, as this would imply a robust system of trust that does not currently exist.
+
+'Regardless of... the vetting process' is highlighted to show that both vetted and new nodes may be disqualified.
+
+> After a storage node is disqualified, the node must go back through the entire vetting process again. If the node decides to start over with a brand-new identity, the node must restart the vetting process from the beginning (in addition to generating a new nodeID via the proof of work system). This strongly disincentivizes storage nodes from being cavalier with their reputation.
+
+Further, the node will be demonetized.
+
+> Provided the storage node hasn’t been disqualified, the storage node will be paid by the Satellite for the data it has stored over the course ofthe month, per the Satellite’s records.
+
+A disqualified SNO should quickly stop particiapting with a satellite it is disqualified and demonetized on.  It may remain in Kademlia and potentially found during Overlay Cache Discover, as the kademlia network supports multiple satellites.
+
+One option that may be allowed for disqualified storage nodes is a Graceful Exit.  [Storage Node Payment and Incentives for V3](https://docs.google.com/document/d/1-Pxzk-ad-0QtF6nnTwfgzk8e_-XbBNSDxuRvnbd0QL8/edit#heading=h.rz1ehm5mbeuz) describes this feature:
+
+> When a node operator wants to leave the network, if they just shut down the node, the unavailability of the data on the network can contribute to repair costs.  If instead, the node triggers a function to call a Satellite, request new storage nodes to store the pieces stored on the node being shut down, then directly upload those pieces to the new nodes, file repair would be avoided.
+
+## Design
+
+### Use of Disqualified Nodes
+
+Disqualified nodes may be used during download of typical downloads from uplinks or via repair.  Therefore, their IP must be tracked in the overlay cache.
+
+Disqualified nodes may not be used for upload, therefore they should not be returned from node selection proceses of any sort.  There is no reason to update the statistics of disqualified nodes.
+
+The list of disqualified nodes should change infrequently, but could grow large over time.
+
+### Handling Disqualified Nodes
+
+Disqualification can be handled in our existing SQL implementation by adding a `disqualified` column to the `nodes` table:
+
+```sql
+CREATE TABLE nodes (
+  id bytea NOT NULL,
+  ...
+  disqualified boolean NOT NULL,
+  PRIMARY KEY ( id )
+);
+```
+
+Existing SQL queries employing logic such as `WHERE audit_success_ratio >= $2 AND uptime_ratio >= $3` would change to `WHERE disqualified = FALSE`.
+
+Existing calls to the DBX `UpdateNodeInfo()` method must set `disqualified` if appropriate.  This may be best accomplished by a database trigger or stored procedure.
```

Comment:
A database trigger will be database specific code. This will make it harder to switch to a different solution later. The nodes table is creating a high load on the database. We might want to store it somewhere else which would makes the database trigger useless.

Could we do it in the code? At the moment the audit job would like to write back the results we can check if the node is now disqualified. Same for uptime checks.

Reply:
Good point.  I am a little worried about the extra db locking of getting those values from the database before setting them again... but we can write Postgres SQL at least that should be a single statement.  Other implementations might not even be SQL. Will update.

Reply:
updated